### ISSUE TITLE
[Boost 1.71] Clang crashes when compiling applications on latest docker images for ubuntu focal

### ISSUE THREAD

**Issue by romainhild** [feelpp/feelpp#1570]
When using the latest image for ubuntu focal, clang crashes when trying to compile `feelpp_qs_adr_3d`, with errors such as
```
1.      <invalid loc> <Spelling=/opt/feelpp/feelpp/quickstart/qs_adr.cpp:98:42>: current parser token ' _rhs=l,'
2.      <invalid>: parsing namespace 'Feel'
...
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f12e68650b3]
/usr/bin/clang++-10(_start+0x2e)[0x40d7ce]
clang: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 10.0.0-4ubuntu1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
```

The dockerfile used is the following:
```
FROM ubuntu:focal

USER root

ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update \
    && apt-get install -y \
        vim \
        ssh \
        sudo \
        wget \
        software-properties-common \
        build-essential \
        git \
        clang-10 \
        cmake \
        libopenmpi-dev \
        libpython3-dev \
        libhdf5-dev \
        libhdf5-openmpi-dev \
        python3-sympy \
        python3-mpi4py \
        python3-distutils \
        libboost1.71-dev \
        libboost-date-time1.71-dev \
        libboost-filesystem1.71-dev \
        libboost-regex1.71-dev \
        libboost-system1.71-dev \
        libboost-program-options1.71-dev \
        libboost-test1.71-dev \
        libboost-mpi1.71-dev \
        libboost-iostreams1.71-dev \
        pkg-config \
        libann-dev \
        libfftw3-mpi-dev \
        libglpk-dev \
        libxml2-dev \
        petsc-dev \
        slepc-dev \
        libopenturns-dev \
        libgsl-dev \
        libcurl4-openssl-dev \
        libcln-dev \
        swig \
        ruby \
        libgmsh-dev

WORKDIR /opt
RUN git clone https://github.com/feelpp/feelpp.git
RUN mkdir feelpp/build # && cd feelpp/build
WORKDIR /opt/feelpp/build
RUN CXX=clang++-10 cmake .. -DCMAKE_BUILD_TYPE=Release -DFEELPP_ENABLE_FMILIB=OFF -DFEELPP_ENABLE_QUICKSTART=ON -DFEELPP_ENABLE_MMG=OFF
RUN cd feelpp/quickstart && make -j 10 feelpp_qs_adr_3d
RUN make install
```

**Comment by prudhomm** [comment]
there are  bug reports on launchpad here  
https://bugs.launchpad.net/ubuntu/+source/llvm-toolchain-11/+bug/1910863
and
https://bugs.launchpad.net/ubuntu/+source/llvm-toolchain-10/+bug/1910932

**Comment by prudhomm** [comment]
this is strange because on buildkite and debian builds we don't see these issues

**Comment by romainhild** [comment]
yes but these bugs are for 20.10, focal is 20.04 right ?
This is strange, because feelpp has been compiled on focal at some points before.

**Comment by prudhomm** [comment]
it currently compiles with focal, see https://buildkite.com/feelpp/feelpp/builds?branch=develop

**Comment by prudhomm** [comment]
it is now in bugs.llvm.org https://bugs.llvm.org/show_bug.cgi?id=48789

**Comment by romainhild** [comment]
This issue is caused by boost 1.71
Replacing it by boost 1.67 solves the issue.

**Comment by prudhomm** [comment]
It seems it might be due to boost parameter. Let's try includinv a working version of boost parameter into feelpp

**Comment by vincentchabannes** [comment]
seems to have the same issue on macos with clang10 and boost 1.71 or boost 1.75 (ok with boost 1.69).
A basic application generates the error if we use : `#include <feel/feel.hpp>`.
No error if include header by the hand, for example these one : 
```
#include <feel/feelcore/environment.hpp>
#include <feel/feeldiscr/pchv.hpp>
#include <feel/feelfilters/geotool.hpp>
#include <feel/feelfilters/exporter.hpp>
#include <feel/feelvf/vf.hpp>
#include <feel/feelts/newmark.hpp>
 ```

**Comment by vincentchabannes** [comment]
we can see all headers includes with clang and gcc by adding -H option

**Comment by metivett** [comment]
Note: this issue is not actually closed. I therefore re-open it for tracking. For example, simple doc code (like `feelpp_doc_laplacian` from `doc/manual/laplacian/laplacian.cpp` still fails to compile and makes clang crash (memory overload).
The proposed workaround for the moment is to not include `<feel/feel.hpp>` but rather use specific includes like 
```
#include <feel/feelcore/environment.hpp>
#include <feel/feelfilters/loadmesh.hpp>
#include <feel/feelfilters/exporter.hpp>
#include <feel/feeldiscr/pch.hpp>
#include <feel/feelvf/vf.hpp>
```

**Comment by prudhomm** [comment]
same issue as in #1697 
possible fixes are
- remove unneeded headers and reorder
-  #1473

**Comment by prudhomm** [comment]
this has been fixed with the introduction of napp
it will be closed automatically when 110 is released
fixes #1570
